src/lib.rs

```diff
@@ -13,24 +29,69 @@ use heck::{ToPascalCase, ToSnakeCase};
 use pad::PadAdapter;
 use std::{
     collections::{BTreeMap, BTreeSet},
-    io::{BufWriter, Write},
+    fmt::Display,
+    io::{self, BufWriter, Write},
 };
+use typed_builder::TypedBuilder;
 
 mod includes;
 mod keywords;
 mod pad;
 
+/// Code generator configuration. See module-level docs for an example.
+#[derive(TypedBuilder)]
+#[non_exhaustive]
+pub struct Config<'a> {
+    /// Name of the dbc-file. Used for generated docs only.
+    pub dbc_name: &'a str,
+
+    /// Raw bytes of a dbc-file.
+    pub dbc_content: &'a [u8],
+
+    /// (optional) Print debug info to stdout while generating code.
+    #[builder(default)]
+    pub debug_prints: bool,
+
+    /// (optional) `impl Debug` for generated types.
+    #[builder(default)]
+    pub impl_debug: FeatureConfig<'a>,
+
+    /// (optional) `impl Arbitraty` for generated types.
+    #[builder(default)]
+    pub impl_arbitrary: FeatureConfig<'a>,
+
+    /// (optional) Whether to validate min and max values for signals.
+    #[builder(default)]
+    pub check_ranges: FeatureConfig<'a>,
```

Comment:
maybe some features should have different default values?

Reply:
Yeah, i check_ranges makes sense to be enabled by default